From 6fd294018bbeac241274b7645498e619e598ce3c Mon Sep 17 00:00:00 2001
From: Guillem Jover <guillem@hadrons.org>
Date: Fri, 1 Mar 2024 00:18:42 +0100
Subject: [PATCH libaio 10/26] Remove unused raw_syscall.c

The ia64_aio_raw_syscall assembler function was replaced by the
__ia64_raw_syscall in commit a4984400f1d40c3cd9353ee2d817271c0fa3b4bd.

The hand-crafted syscalls for all architectures then got removed
in commit 97fd3fc0195500e616e34047cba4846164c411d9.

This file has not actually been used for a while, so we can drop it.

Signed-off-by: Guillem Jover <guillem@hadrons.org>
---
 src/Makefile      |  3 ---
 src/raw_syscall.c | 19 -------------------
 2 files changed, 22 deletions(-)
 delete mode 100644 src/raw_syscall.c

diff --git a/src/Makefile b/src/Makefile
index 8a9bfb4..387b57f 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -30,9 +30,6 @@ libaio_srcs += io_queue_wait.c io_queue_run.c
 libaio_srcs += io_getevents.c io_submit.c io_cancel.c
 libaio_srcs += io_setup.c io_destroy.c io_pgetevents.c
 
-# internal functions
-libaio_srcs += raw_syscall.c
-
 # old symbols
 libaio_srcs += compat-0_1.c
 
diff --git a/src/raw_syscall.c b/src/raw_syscall.c
deleted file mode 100644
index c3fe4b8..0000000
--- a/src/raw_syscall.c
+++ /dev/null
@@ -1,19 +0,0 @@
-#include "syscall.h"
-
-#if defined(__ia64__)
-/* based on code from glibc by Jes Sorensen */
-__asm__(".text\n"
-	".globl	__ia64_aio_raw_syscall\n"
-	".proc	__ia64_aio_raw_syscall\n"
-	"__ia64_aio_raw_syscall:\n"
-	"alloc r2=ar.pfs,1,0,8,0\n"
-	"mov r15=r32\n"
-	"break 0x100000\n"
-	";;"
-	"br.ret.sptk.few b0\n"
-	".size __ia64_aio_raw_syscall, . - __ia64_aio_raw_syscall\n"
-	".endp __ia64_aio_raw_syscall"
-);
-#endif
-
-;
-- 
2.43.0

